types/src/worker.rs

```diff
@@ -18,3 +20,55 @@ pub enum WorkerMessage<PublicKey: VerifyingKey> {
 
 #[derive(Debug, Serialize, Deserialize)]
 pub struct ClientBatchRequest(pub Vec<BatchDigest>);
+
+/// Indicates a serialized `WorkerMessage::Batch` message.
+pub type SerializedBatchMessage = Vec<u8>;
+
+/// Hashes a serialized batch message without deserializing it into a batch.
+///
+/// See the test `test_batch_and_serialized`, which guarantees that the output of this
+/// function remains the same as the [`Hash::digest`] result you would get from [`Batch`].
+/// See also the micro-benchmark `batch_digest`, which checks the performance of this is
+/// identical to hashing a serialized batch.
+///
+/// TODO: remove the expects in the below, making this return a `Result` and correspondingly
+/// doing error management at the callers.
```

Comment:
Do we have an issue for this?

Reply:
https://github.com/MystenLabs/narwhal/issues/268